test/conformance/kernel/urKernelSetArgPointer.cpp

```diff
@@ -33,27 +41,39 @@ TEST_P(urKernelSetArgPointerTest, SuccessHost) {
     ASSERT_NE(allocation, nullptr);
 
     ASSERT_SUCCESS(urKernelSetArgPointer(kernel, 0, allocation));
+    ASSERT_SUCCESS(urKernelSetArgValue(kernel, 1, sizeof(data), &data));
```

Comment:
As far as I can tell from the spec and the IR you get from the fill_usm test program, these *should* work - but none of them do with the cuda plugin. I'm still looking into it but the test code is sound I think

Reply:
What's not working about this on the CUDA adapter?

Reply:
So I'm finding it works fine if I do this
```
ASSERT_SUCCESS(urKernelSetArgPointer(kernel, 0, &allocation));
```
instead of just passing the `allocation` pointer. This makes sense looking at how the plugin implements passing the arg through to be stored in its kernel struct; but it doesn't match the UR spec, the pi header or the analogous CL entry point. Currently trying to figure out if this is de-facto how the sycl runtime is using the entry point from the original plugin and we just need to update the spec to reflect that, or if there's something weirder happening

Reply:
Ah, yeah. We deffo want consistency with this. I wonder if that's a holder over from PI which is a holder over from OpenCL... Do OpenCL implenmetations of `clSetKernelArg` do `&mem_object`?

Reply:
Alright what I've just pushed is fully working. I think I've gathered enough insight into how the runtime and various plugins are using this stuff to justify changing the signature of `urKernelSetArgPointer` to take `void**` instead of just `void*`. Should be a nfc but it better represents how the entry point is supposed to be used in conjunction with the various USM alloc entry points.